exporter/collector/metricsexporter.go

```diff
@@ -925,3 +1008,52 @@ func mapMetricPointKind(m pdata.Metric) (metricpb.MetricDescriptor_MetricKind, m
 	}
 	return kind, typ
 }
+
+func (me *MetricsExporter) exportTimeSeriesRunner() {
+	defer me.goroutines.Done()
+	me.batchTimeoutTimer = time.NewTimer(batchTimeout)
+	for {
+		select {
+		case <-me.shutdownC:
+			for {
+				// We are shutting down. Publish all the pending
+				// items on the channel before we stop.
+				select {
+				case item := <-me.timeSeriesC:
+					me.processItem(item)
+				default:
+					goto DONE
+				}
+			}
+		DONE:
+			for len(me.pendingTimeSerieses) > 0 {
+				me.exportPendingTimeSerieses()
+			}
+			// Return and continue graceful shutdown.
+			return
+		case item := <-me.timeSeriesC:
+			me.processItem(item)
+		case <-me.batchTimeoutTimer.C:
+			for len(me.pendingTimeSerieses) > 0 {
+				me.exportPendingTimeSerieses()
+			}
+			me.batchTimeoutTimer.Reset(batchTimeout)
```

Comment:
Should we reset the timer before exporting pending time series so that each batch only contains 10s of metrics, and rather than 10s + createTimeSeries latency?

Reply:
Done.